### PR TITLE
Began method to convert string matrix to ArrayList<ArrayList<Fraction>>

### PR DESCRIPTION
**Pull request by mohtashim-ali**

### CODE REVIEW

**Comment by NofelYazdani** [comment]
Impressive work. Smart use of recursion helper. UI should be nearly done now, good job!

File: src/main/java/UseCase/BuildMatrix.java [hunk 1]
```diff
@@ -7,54 +7,52 @@
...
+        ArrayList<ArrayList<Fraction>> matri = new ArrayList<>();
+        while (x < matrix.length){
+            if (Objects.equals(matrix[x].toString(), "[")){
+                x++;
+                ArrayList<Fraction> inner = new ArrayList<>();
+                while (!matrix[x].toString().equals("]")){
+                    if (!matrix[x].toString().equals(",") && !matrix[x].toString().equals("]")){
+                        Fraction test = new Fraction(Integer.parseInt(matrix[x]), Integer.parseInt(matrix[x+2]));
+                        inner.add(test);
+                        x++;
```
Comment (krhobeck) [inline src/main/java/UseCase/BuildMatrix.java:38]: should just be x+=3;

File: src/main/java/UseCase/BuildMatrix.java [hunk 2]
```diff
@@ -7,54 +7,52 @@
...
-        ArrayList<ArrayList<Fraction>> matrix = new ArrayList<>();
+    public ArrayList<ArrayList<Fraction>> helper(String[] matrix){
+
+        int x  = 1;
+        ArrayList<ArrayList<Fraction>> matri = new ArrayList<>();
+        while (x < matrix.length){
+            if (Objects.equals(matrix[x].toString(), "[")){
+                x++;
+                ArrayList<Fraction> inner = new ArrayList<>();
+                while (!matrix[x].toString().equals("]")){
```
Comment (krhobeck) [inline src/main/java/UseCase/BuildMatrix.java:34]: if you're going to use 'matrix[x].toString()' multiple times make it a variable

**Comment by HamzaShariff** [comment]
Great job Mohtashim! This makes it much easier for the user to input a matrix. This will also be essential to our Monday presentation.

File: src/main/java/UseCase/BuildMatrix.java [hunk 3]
```diff
@@ -7,54 +7,52 @@
...
+                    if (!matrix[x].toString().equals(",") && !matrix[x].toString().equals("]")){
+                        Fraction test = new Fraction(Integer.parseInt(matrix[x]), Integer.parseInt(matrix[x+2]));
```
Comment (krhobeck) [inline src/main/java/UseCase/BuildMatrix.java:36]: also create variables for 'Integer.parseInt(matrix[x]) and Integer.parseInt(matrix[x+2]) to make it cleaner

**Review by krhobeck — CHANGES_REQUESTED** [review]
Fix these before pushing

File: src/main/java/UseCase/BuildMatrix.java (hunk 3 above)
Comment (mohtashim-ali) [reply src/main/java/UseCase/BuildMatrix.java]: I agree, change is now pushed to the branch

File: src/main/java/UseCase/BuildMatrix.java (hunk 1 above)
Comment (mohtashim-ali) [reply src/main/java/UseCase/BuildMatrix.java]: Yup thanks for catching that

File: src/main/java/UseCase/BuildMatrix.java (hunk 2 above)
Comment (mohtashim-ali) [reply src/main/java/UseCase/BuildMatrix.java]: I will look into this later as the x value changes

**Comment by mohtashim-ali** [comment]
Since I have my team's approval and feedback, and I have Made the changes I will merger this to finish off the project

**Comment by mohtashim-ali** [comment]
Fixed Conflicts

**Comment by AchrafBoufer** [comment]
Looks like you fixed the issue, good job!

**Pull request merged**